### PR TITLE
armstdc: add limits.h header

### PR DESCRIPTION
**Pull request by galak**
Add limits.h header that defines PATH_MAX as this is not defined by the toolchain headers.

### CODE REVIEW

File: lib/libc/armstdc/include/limits.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#ifndef ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_LIMITS_H_
+#define ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_LIMITS_H_
+
+#include_next <limits.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define PATH_MAX 256
```
Comment (keith-packard) [inline lib/libc/armstdc/include/limits.h:16]: Where did this particular value come from? picolibc uses 1024, fwiw.

Comment (galak) [reply lib/libc/armstdc/include/limits.h:16]: I grabbed the value from lib/libc/minimal/include/limits.h

Comment (galak) [reply lib/libc/armstdc/include/limits.h:16]: Updated commit message to explain where the value came from.

Comment (keith-packard) [reply lib/libc/armstdc/include/limits.h:16]: ok, seems fine to me, although a bit small by comparison with other systems.

**Review by cfriedt — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**